apis/Google.Cloud.Spanner.Data/docs/configuration.md

```diff
@@ -0,0 +1,180 @@
+# Configuration
+
+Many applications will only ever need to configure Spanner in terms
+of the database to connect via the connection string. (The database
+name consists of a project ID, instance ID and database ID.)
+
+Beyond that, configuration is largely for the sake of diagnostics
+and performance tuning. There are two main source for configuration:
+
+- The connection string provided to `SpannerConnection`. Options
+  here are also exposed as properties on
+  `SpannerConnectionStringBuilder`
+- The session pool options, as exposed via
+  `Google.Cloud.Spanner.V1.SessionPoolOptions`.
+
+The connection string options are [specified in this
+document](connection_string.md). The remaining options relate to the
+session pool, which handles acquiring, refreshing and evicting
+sessions. Spanner sessions are relatively expensive (in terms of
+time) to create, so it's important that they are handled
+appropriately.
+
+## Session pool options explained
+
+`SessionPoolOptions` exposes the following properties:
+
+**MaximumActiveSessions**
+
+Default value: 100
+
+The maximum number of sessions that can be active per database. An
+active session is one that has been acquired but not yet released
+back to the pool. Broadly speaking, this is equivalent to the number
+of concurrent Spanner-based operations your application will be able
+to perform.
+
+Applications may wish to change this setting if they require high
+levels of concurrency.
+
+**MinimumPooledSessions**
+
+Default value: 10
+
+The minimum number of sessions to maintain in the pool of available
+sessions, on a per-database basis. If the number of pooled sessions
+falls below this number, more sessions are added automatically,
+unless that would require more than the configured maximum active
+session count.
```

Comment:
Maybe be more explicit in saying that in practice, MinimumPooledSessions is upper bounded by MaximumActiveSessions.

Reply:
I've added a sentence - see what you think.